pkg/graph/walker.go

```diff
@@ -26,97 +25,43 @@ func WalkRewrite(rewrite *core.UsersetRewrite, handler WalkHandler) interface{}
 	case *core.UsersetRewrite_Exclusion:
 		return walkRewriteChildren(rw.Exclusion, handler)
 	default:
-		panic(fmt.Sprintf("unknown type of rewrite operation in walker: %T", rw))
+		return nil, spiceerrors.MustBugf("unknown type of rewrite operation in walker: %T", rw)
 	}
 }
 
 // HasThis returns true if there exists a `_this` node anywhere within the given rewrite. If
 // the rewrite is nil, returns false.
-func HasThis(rewrite *core.UsersetRewrite) bool {
-	result := WalkRewrite(rewrite, func(childOneof *core.SetOperation_Child) interface{} {
+func HasThis(rewrite *core.UsersetRewrite) (bool, error) {
+	result, err := WalkRewrite(rewrite, func(childOneof *core.SetOperation_Child) interface{} {
 		switch childOneof.ChildType.(type) {
 		case *core.SetOperation_Child_XThis:
 			return true
 		default:
 			return nil
 		}
 	})
-	return result != nil && result.(bool)
+	return result != nil && result.(bool), err
 }
 
-func walkRewriteChildren(so *core.SetOperation, handler WalkHandler) interface{} {
+func walkRewriteChildren(so *core.SetOperation, handler WalkHandler) (interface{}, error) {
 	for _, childOneof := range so.Child {
 		vle := handler(childOneof)
 		if vle != nil {
-			return vle
+			return vle, nil
 		}
 
 		switch child := childOneof.ChildType.(type) {
 		case *core.SetOperation_Child_UsersetRewrite:
-			rvle := WalkRewrite(child.UsersetRewrite, handler)
-			if rvle != nil {
-				return rvle
+			rvle, err := WalkRewrite(child.UsersetRewrite, handler)
+			if err != nil {
+				return nil, err
 			}
-		}
-	}
-
-	return nil
-}
-
-type OperationPath = []uint32
-
-// FindOperation finds the operation at the given path in the rewrite, if any.
-func FindOperation[T any](rewrite *core.UsersetRewrite, opPath OperationPath) *T {
```

Comment:
dead code that was recursively calling each other?

Reply:
Yes, I meant to use it for reachability analysis but then it got un-used and I never removed it. I figured I'd just remove it here since it isn't used aynmore